src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -987,42 +987,46 @@ private void DecodeInterceptsLocationAttribute(DecodeWellKnownAttributeArguments
             }
 
             var syntaxTrees = DeclaringCompilation.SyntaxTrees;
-            SyntaxTree? matchingTree = null;
-            // PROTOTYPE(ic): we need to resolve the paths before comparing (i.e. respect /pathmap).
-            // At that time, we should look at caching the resolved paths for the trees in a set (or maybe a Map<string, OneOrMany<SyntaxTree>>).
-            // so we can reduce the cost of these checks.
-            foreach (var tree in syntaxTrees)
+            var matchingTrees = DeclaringCompilation.GetSyntaxTreesByMappedPath(attributeFilePath);
+            if (matchingTrees.Count == 0)
             {
-                if (tree.FilePath == filePath)
+                var referenceResolver = DeclaringCompilation.Options.SourceReferenceResolver;
```

Comment:
nit: consider moving closer to usage

Reply:
Because it's used in each subsequent branch in this block, I felt more comfortable declaring it where it is currently, instead of after the statement declaring `unmappedMatch`, for example.

I'm line-breaking some of the `diagnostics.Add` calls to make the usages easier to spot.